docs/compiler-exceptions.rst

```diff
@@ -0,0 +1,277 @@
+        /usr/lib/python3/dist-packages/apport/report.py:13: DeprecationWarning: the imp module is deprecated in favour of             importlib; see the module's documentation for alternative uses
+        import fnmatch, glob, traceback, errno, sys, atexit, locale, imp
+        vyper.exceptions.FunctionDeclarationException: Function name shadowing a variable name: pendingReturns
+
+    The warning generated in the terminal does not specify any line numbers.  ``pendingReturns`` is named as the identifier
+    used incorrectly to declare the function throwing the error.  
+
+.. py:exception:: InvalidLiteralException
+
+    .. code-block:: python3
+
+        @public
+        def foo():
+            bar: address = 0xdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef
+        
+    Raised when attempting to use a literal value where the type is correct, but the value is still invalid in some way. For example, an address that is not check-summed.
+
+.. py:exception:: InvalidTypeException
+
+    .. code-block:: python3
+
+        bids: map(address, Bid[128])
+        bidCounts: map(adddress, int128)
+
+    The variable type ``address`` is misspelled.  Any word that is not a reserved word, and declares a variable type will 
+    return this error.
+
+    .. code-block:: bash
+
+        $ vyper blind_auction.vy 
+        Error compiling: blind.auction.vy /usr/lib/python3/dist-packages/apport/report.py:13: 
+        DeprecationWarning: the imp module is deprecated in favour of
+        importlib; see the module's documentation for alternative uses
+        import fnmatch, glob, traceback, errno, sys, atexit, locale, imp
+        vyper.exceptions.InvalidTypeException: line 28:15 Invalid base type: adddress
+                 27 bids: map(address, Bid[128])
+            ---> 28 bidCounts: map(adddress, int128)
+            -----------------------^
+                 29
+
+    The terminal returns a compiling error warning.  Reading the entire warning is critical to understanding exactly what
+    is causing the error.  The message displays the line numbers that contain the error.  In this example ``map(_KeyType, _ValueType)`` cannot compile because the type ``address`` is misspelled.
+
+.. py:exception:: JSONError
+
+    Vyper has the ability to pass information back and forth using JSON.  If you are using JSON and receiving a JSON error then you can find out more details about the error at `Oracle JSON Errors <https://docs.python.org/3/tutorial/errors.html>`_.
+
+.. py:exception:: NonPayableViolationException
+
+    .. code-block:: python3
+
+        @private
+        def _foo():
+            bar: uint256 = msg.value
+       
+    Raised when attempting to access ``msg.value`` from within a private function.
+
+.. py:exception:: ParserException
+
+    .. code-block:: python3
+
+        @private
+        def foo(a: address = msg.sender):
+            pass
+        vyper.exceptions.ParserException: line 3:21 msg.sender not allowed in private functions.
```

Comment:
Similar to the requested change for `TypeMismatchError`, this should also be split into two code blocks with the description moved in between.

Reply:
fixed